### ISSUE TITLE
ceiling fan disconnects

### ISSUE THREAD

**Issue by vmarks** [zwaldowski/homebridge-satellite-fan#7]
Hi, 
as I run homebridge, I see status events as I change the settings on the fan. I also frequently see the fan as disconnected. Can you give me any advice for what to do to prevent disconnects? I'd like to keep the fan responsive. (There are times where I try and toggle the light and it doesn't change, even though homebridge and homekit believe it has - until coalesced picks up that it hasn't actually changed. Or at least, that's my understanding of what's happening. Where should I look next?

[06/06/2018, 10:46:30] [Ceiling Fan] Light on: true
[06/06/2018, 10:46:37] [Ceiling Fan] Light on: false
noble warning: unknown peripheral 5c313ec5b57e
[06/06/2018, 10:49:32] [Ceiling Fan] Disconnected
[06/06/2018, 10:49:40] [Ceiling Fan] Ready

**Comment by zwaldowski** [comment]
Yeah, this has been one of the biggest struggles for me. I assume your hardware is an RPi, or similar? My understanding is that that it combines its Wi-Fi and Bluetooth into a single module; this in itself should not cause disconnects, but your Bluetooth-handling code has to be more forgiving on its timing, which noble/noble appears not to be. In the fullness of time I definitely want a fix, I’m just kinda stuck at the moment.

**Comment by vmarks** [comment]
Hi, thanks for coming back to this.

I'm on an RPi 2, using a USB IOGear bluetooth adapter. Wi-Fi is not configured, and I'm using ethernet for the connection to the network.

**Comment by vmarks** [comment]
What should I use instead of Rpi? I’ve got some old AMD powered machines here, I could use those if you thought they would work any better?

**Comment by zwaldowski** [comment]
I don't think it's down to the specific hardware. You could definitely try it, but I can't say off the top of my head it'd be more reliable. My understanding is that it's an implementation issue with noble/noble. I would love to be able to swap out the internals, I just don't have that capability at the moment.

**Comment by iRayanKhan** [comment]
can you paste your config please thanks

**Comment by vmarks** [comment]
`{
    "bridge": {
        "name": "Homebridge",
        "username": "CC:22:3D:E3:CE:30",
        "port": 51826,
        "pin": "031-45-154"
    },
"accessories": [ {
      "name": "Ceiling Fan",
      "accessory": "satellite-fan",
      "address": "5C:31:3E:C5:B5:7E",
      "device": { 
        "manufacturer": "Harbor Bay",
        "model": "00000", 
        "serial": "000000000000",
        "light": true
      },
      "ble": {
        "prefix": 64,
        "serviceUUID": "539C6813-61A0-2137-4F79-BF1A11984790", 
        "writeCharacteristicUUID": "539C6813-61A1-2137-4F79-BF1A11984790",
        "notifyCharacteristicUUID": "539C6813-61A2-2137-4F79-BF1A11984790"
      }
  } ]
}


`

**Comment by vmarks** [comment]
I erased my raspberry pi, still using an Iogear BLE USB, no wifi, reinstalled everything from scratch, using node 8.xx.

[2018-9-30 13:32:18] [Ceiling Fan] Ready
noble warning: unknown peripheral 5c313ec5b57e
[2018-9-30 13:32:57] [Ceiling Fan] Disconnected
[2018-9-30 13:32:59] [Ceiling Fan] Disconnected

still happens. Not sure why, or what can be done to improve it. Any tips are welcome.
